scripts/do-release.sh

```diff
@@ -67,15 +67,18 @@ function _update_go_mod() {
 
 function update_go_mod() {
     local target="$1"
-    if [[ ! -f projects/${project}/go.mod ]]; then
-        return 1
-    fi
 
-    # Run in subshell so we don't change the working directory even on failure
-    (
-        cd "projects/${project}"
-        dryrun _update_go_mod
-    )
+    shopt -s globstar
+    for gomod in projects/${project}/**/go.mod; do
+        dir="${gomod%/*}"
+        if [ -d "$dir" ]; then
+            # Run in subshell so we don't change the working directory even on failure
+            (
+                cd "$dir"
+                dryrun _update_go_mod
+            )
+        fi
```

Comment:
```suggestion
        if [ ! -d "$dir" ]; then
            echo "WARN: Found ${gomod} but ${dir} is not a directory, will not update it.
            continue
        fi

        # Run in subshell so we don't change the working directory even on failure
        ( cd "$dir"; dryrun _update_go_mod; )
```

Reply:
The point of the test is to catch projects with no `go.mod` at all — `dir` is then `**/go.mod`, and `-d "**"` fails. I don’t think there’s any point logging a warning in such circumstances ;-).

Reply:
In that case don't we want to preserve original functionality where it would fail if project had no go.mod?

Reply:
And I still prefer "test on exceptional cases" writing style in this case as its easier to understand, instead of "expect preconditions".

Also the way you originally suggest just silently ignores it which is not what we want generally as it makes debugging failures much harder later on.